File: src/test/java/com/znsio/teswiz/screen/autoscroll/AutoScrollScreen.java
package com.znsio.teswiz.screen.autoscroll;

import com.znsio.teswiz.entities.Direction;

public abstract class AutoScrollScreen {

    public static AutoScrollScreen get() {
        return com.znsio.teswiz.screen.ScreenRegistry.getScreen(AutoScrollScreen.class);
    }

    public abstract AutoScrollScreen goToDropdownWindow();

    public abstract AutoScrollScreen scrollInDynamicLayer(Direction direction);

    public abstract boolean isScrollSuccessful();
}

File: src/test/resources/playwright/screens/autoscroll/auto-scroll.screen.ts
import { unsupportedScreenAction } from "../screen-route.ts";

export function goToDropdownWindow(): object {
  return unsupportedScreenAction(
    "AutoScroll is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}

export function scrollInDynamicLayer(): object {
  return unsupportedScreenAction(
    "AutoScroll is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}

export function isScrollSuccessful(): object {
  return unsupportedScreenAction(
    "AutoScroll is not supported on web for WEB_ENGINE=playwright-ts.",
  );
}
